Makefile

```diff
@@ -0,0 +1,26 @@
+MAKEFILE := $(lastword $(MAKEFILE_LIST))
+PWD := $(patsubst %/,%,$(dir $(abspath $(MAKEFILE))))
+
+DASHBOARD_TEMPLATE_NAME = quickstart-cloudwatch-dashboard.yaml
+DASHBOARD_CONFIG_NAME = dashboard_config.json
+
+DASHBOARD_CONFIG_FILE := $(PWD)/templates/config/$(DASHBOARD_CONFIG_NAME)
+DASHBOARD_TEMPLATE_FILE := $(PWD)/templates/config/$(DASHBOARD_TEMPLATE_NAME).template
+DASHBOARD_TEMPLATE := $(PWD)/templates/$(DASHBOARD_TEMPLATE_NAME)
+
+define prettyecho
+        $(if $(TERM),
+                @tput setaf $2
+                @echo $1
+                @tput sgr0,
+                @echo $1)
+endef
+
+create_dashboard_template:
+		@echo
+		$(call prettyecho, "Reading contents of - $(DASHBOARD_CONFIG_FILE) and replacing marker in $(DASHBOARD_TEMPLATE_FILE)", 13)
```

Comment:
I am of two minds with this. I applaud that we are limiting and minimising duplication but I am not sure if `Makefile` is the best format for template processing. Wouldn't be easier and more maintainable to have a small python script that will process `jinja` template so it consistent with our docker repo? I think anybody who will work with this particular file would have a pretty deep understanding of our deployment and it is semi-safe to assume they will have python installed.

Reply:
I was tempted to reuse the Jinja approach here, but didn't follow through on the Jinja approach as it would have been too heavy for this. Getting Jinja would mean introducing python to the repo, with requirements.txt. It was easier to do so in the ansible automation repo (which uses `make` but invokes python instead of bash scripting here), as python was already a pre-requisite.

I'm happy to change use python instead; my concern is that it may not add too much value for a single line replacement that we want to accomplish here.

Reply:
For what it's worth I think the jinja would be more readable. I don't really understand Makefile and its voodoo.

Reply:
IMO - The only thing that using python would change is that this call 
```
cat $(DASHBOARD_TEMPLATE_FILE) | sed -e s~DASHBOARD_CONFIG~'$(shell cat $(DASHBOARD_CONFIG_FILE) | tr -s "[:space:]" | tr -d '\t' | tr -d '\n')'~g > $(DASHBOARD_TEMPLATE)
```
will change to
```
python generate_dashboard_template <json_config_file> > $(DASHBOARD_TEMPLATE)
```

`make` is just the runner. We'll be using `jinja` instead of `sed` and `tr` magic :)